pkg/kubernetes/jobs.go

```diff
@@ -74,20 +76,53 @@ func (k *KubeJobManager) StartJobCleaner(ctx context.Context, wg *sync.WaitGroup
 }
 
 // LaunchJob launches a Kubernetes job for a given Buildkite job ID
-func (k *KubeJobManager) LaunchJob(uuid string) error {
+func (k *KubeJobManager) LaunchJob(job *buildkite.Job) error {
 	var err error
+	var t batchv1.Job
+	uuid := *job.ID
 
 	jobLabels := make(map[string]string)
 	jobLabels["kubekite-managed"] = "true"
 	jobLabels["kubekite-org"] = k.org
 	jobLabels["kubekite-pipeline"] = k.pipeline
 
-	t := k.jobTemplate
+	err = json.Unmarshal(k.jobTemplate, &t)
```

Comment:
Since we're editing the object down below, those changes were affecting the stored object. So if one step had an image set, and the next one didn't, it would use the prior image as a result of modifying `k.jobTemplate`'s inner structs. The alternative was to create a deep copy each time, but I decided I'd rather Unmarshall each time rather than deep copy the struct.